src/components/ScreenWrapper.js

```diff
@@ -46,14 +53,19 @@ class ScreenWrapper extends React.Component {
         this.unsubscribe = KeyboardShortcut.subscribe('Escape', () => {
             this.props.navigation.goBack();
         }, [], true);
+
+        this.unsubscribeTransitionEnd = this.props.navigation.addListener('transitionEnd', () => {
+            this.props.onTransitionEnd();
+        });
     }
 
     componentWillUnmount() {
-        if (!this.unsubscribe) {
-            return;
+        if (this.unsubscribe) {
+            this.unsubscribe();
+        }
```

Comment:
I know you didn't add this one (it was me!) but since we are adding a new listener can we maybe make the name for this more specific e.g. 

```js
this.unsubscribeEscapeKey();
```